packages/api/tests/lambdas/test-cleanExecutions.js

```diff
@@ -1,58 +1,470 @@
+      status: 'completed',
+    }));
+    records.push(fakeExecutionRecordFactory({
+      updated_at: moment().subtract(i, 'days').toDate(),
+      final_payload: '{"a": "b"}',
+      original_payload: '{"b": "c"}',
+      status: 'failed',
+    }));
+  }
+  const model = new ExecutionPgModel();
 
-class executionModel {
-  removeOldPayloadRecords(...value) {
-    return removeRecordsStub(...value);
+  const pgRecords = await model.insert(t.context.knex, records, '*');
+  for (const record of pgRecords) {
+    await t.context.esClient.client.index({
+      body: await translatePostgresExecutionToApiExecution(record),
+      id: record.cumulus_id,
+      index: t.context.esIndex,
+      type: 'execution',
+      refresh: true,
+    });
   }
-}
+  t.context.execution_cumulus_ids = pgRecords.map((record) => record.cumulus_id);
+});
 
-test.beforeEach(() => {
-  process.env.completeExecutionPayloadTimeoutDisable = false;
-  process.env.nonCompleteExecutionPayloadTimeoutDisable = false;
+test.afterEach.always(async (t) => {
+  await destroyLocalTestDb({
+    knex: t.context.knex,
+    knexAdmin: t.context.knexAdmin,
+    tesetDbName: t.context.testDbName,
+  });
+  await cleanupTestIndex(t.context);
 });
 
-test.serial('Function is called with correct timeout values', async (t) => {
-  process.env.completeExecutionPayloadTimeout = 100;
-  process.env.nonCompleteExecutionPayloadTimeout = 50;
-  await cleanExecutionPayloads(executionModel);
-  t.truthy(removeRecordsStub.calledWith(100, 50, false, false));
+const pgPayloadsEmpty = (entry) => !entry.final_payload && !entry.orginal_payload;
+
+const esPayloadsEmpty = (entry) => !entry.finalPayload && !entry.orginalPayload;
+
+test.serial('handler() handles complete expiration', async (t) => {
+  const env = clone(process.env);
+  process.env = localStackConnectionEnv;
+  process.env.PG_DATABASE = t.context.testDbName;
+  process.env.ES_INDEX = t.context.esIndex;
+  process.env.LOCAL_ES_HOST = 'localhost';
+  let expirationDays = 4;
+  let {
+    completeExpiration: expirationDate,
+  } = getExpirationDates(
+    expirationDays,
+    0,
+    true,
+    false
+  );
+  process.env.completeExecutionPayloadTimeoutDisable = 'false';
+  process.env.nonCompleteExecutionPayloadTimeoutDisable = 'true';
+  process.env.completeExecutionPayloadTimeout = expirationDays;
+
+  await handler();
+  await sleep(5000);
+  const model = new ExecutionPgModel();
+  let massagedPgExecutions = await Promise.all(
+    t.context.execution_cumulus_ids.map(
+      async (cumulusId) => await model.get(t.context.knex, { cumulus_id: cumulusId })
+    )
+  );
+
+  massagedPgExecutions.forEach((massagedExecution) => {
+    if (massagedExecution.updated_at < expirationDate && massagedExecution.status === 'completed') {
+      t.true(pgPayloadsEmpty(massagedExecution));
+    } else {
+      t.false(pgPayloadsEmpty(massagedExecution));
+    }
+  });
+  let massagedEsExecutions = await t.context.searchClient.query({
```

Comment:
I'm wondering if it would be better to split this test out into a couple assertions. Maybe one for PG and one for ES? Just thinking about removing ES in the near future. Might make it easier to just delete a whole test?

Reply:
Also it looks like there's a separate test file for ES executions. Do we need _any_ ES assertions here?